### PR TITLE
add set.remove

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by tpoterba** [comment]
You'll need to add these as class methods on `SetExpression` in `hail.expr.expression.py`

**Comment by tpoterba** [comment]
See here for examples: https://github.com/hail-is/hail/pull/2579

**Comment by tpoterba** [comment]
sorry, this one: https://github.com/hail-is/hail/pull/2575

File: python/hail/expr/expression.py [hunk 1]
```diff
@@ -526,6 +526,18 @@ class SetExpression(CollectionExpression):
     def add(self, x):
         return self._method("add", self._type, x)
 
+    def remove(self, x):
+        """Returns the result of removing the argument from this Set.
+        Examples
+        --------
+        .. doctest:
+
+            >>> eval_expr(f.capture([1,2,3]).to_set.remove(1)
```
Comment (tpoterba) [inline python/hail/expr/expression.py]: should be `to_set()`. Actually, should be `{1, 2, 3}`.

File: python/hail/expr/expression.py [hunk 2]
```diff
@@ -526,6 +526,18 @@ class SetExpression(CollectionExpression):
...
+            {2, 3}
+
```
Comment (tpoterba) [inline python/hail/expr/expression.py]: Needs Parameter and Returns sections

File: python/hail/expr/expression.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/expr/expression.py]: also, the doctest directive is missing a colon on the end: `.. doctest::`

File: src/test/scala/is/hail/methods/ExprSuite.scala [hunk 3]
```diff
@@ -341,6 +341,9 @@ class ExprSuite extends SparkSuite {
     assert(eval[Set[_]]("""iset.flatMap(x => [x].toSet())""").contains(Set(0, 1, 2)))
     assert(eval[Set[_]]("""iset.flatMap(x => [x, x + 1].toSet())""").contains(Set(0, 1, 2, 3)))
     assert(eval[Set[_]]("""iset.add(3)""").contains(Set(0, 1, 2, 3)))
+    assert(eval[Set[_]]("""iset.remove(3)""").contains(Set(0, 1, 2)))
+    assert(eval[Set[_]]("""iset.remove(0)""").contains(Set(1, 2)))
+    assert(eval[Set[_]]("""iset.add(3).remove(0)""").contains(Set(0, 1, 2)))
```
Comment (catoverdrive) [inline src/test/scala/is/hail/methods/ExprSuite.scala]: think you mean Set(1, 2, 3) here?

**Comment by tpoterba** [comment]
Ah, looks like it's failing because eval_expr isn't imported here.

**Comment by tpoterba** [comment]
This PR has a fix: https://github.com/hail-is/hail/pull/2579

**Comment by tpoterba** [comment]
You could also add a `.. testsetup::` directive to import it here.

**Comment by danking** [comment]
@tpoterba I'll wait for #2579 to land

**Pull request merged**